pkg/cmd/backuprestore/backupserver.go

```diff
@@ -61,7 +86,49 @@ func (b *backupServer) Run(ctx context.Context) error {
 	defer cancel()
 	signal.NotifyContext(cCtx, shutdownSignals...)
 
+	if b.enabled {
+		schedule, err := cron.ParseStandard(b.schedule)
+		if err != nil {
+			pErr := fmt.Errorf("error parsing backup schedule %v: %w", b.schedule, err)
+			klog.Error(pErr)
+			return pErr
+		}
+
+		err = b.scheduleBackup(cCtx, schedule)
+		if err != nil {
+			return err
+		}
+	}
+
 	<-ctx.Done()
+	return nil
+}
+
+func (b *backupServer) scheduleBackup(ctx context.Context, schedule cron.Schedule) error {
+	ticker := time.NewTicker(time.Until(schedule.Next(time.Now())))
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			err := b.runBackup()
+			if err != nil {
+				return err
+			}
+			ticker.Reset(time.Until(schedule.Next(time.Now())))
+		case <-ctx.Done():
+			klog.Infof("context deadline has exceeded")
```

Comment:
that could be misleading, because you don't know that this context was a deadline. I think you can safely remove that statement and just leave a `return nil` or so

Reply:
done